addons/web/static/src/views/fields/many2one/many2one_field.js

```diff
@@ -95,13 +96,21 @@ export class Many2OneField extends Component {
         });
         this.computeActiveActions(this.props);
 
+        this.updateRecord = async (value) => {
```

Comment:
could you define it as a "normal" function on the component?
Also, would probably be more natural written like
```js
const changes = { [this.props.name]: value };
if (this.props.update) {
    return this.props.update(changes);
}
return this.props.record.update(changes);
```